libservo: Introduce `SiteDataManager` and `NetworkManager`

janvarga opened the pull request:
This PR introduces an initial `SiteDataManager` abstraction intended
primarily for embedders to inspect and manage site data.

At this stage, `SiteDataManager` supports clearing network cache and
cookies. Cookie related functionality has been moved here from the
former `CookieManager`, which has been renamed to `NetworkManager` to
better reflect its scope.

Additional site data APIs will be added in follow-up PRs.

Testing: Added a new integration test with a TODO comment.

janvarga committed:
libservo: Introduce `SiteDataManager` for embedders

This patch introduces an initial `SiteDataManager` abstraction intended
primarily for embedders to inspect and manage site data.

At this stage, only clearing of network cache is supported. Additional
functionality will be added in follow-up PRs.

Signed-off-by: Jan Varga <jvarga@igalia.com>

janvarga commented on components/servo/site_data.rs:
```diff
@@ -0,0 +1,28 @@
+
+    // TODO: This currently does not wait for cache clearing to complete.
+    pub fn clear_cache(&self) {
+        self.public_resource_threads.clear_cache();
```
I have a WIP patch which adds `GenericCallback` to `ResourceThreads::clear_cache`.

mrobinson reviewed (commented):
Just a question: does it makes sense for the `CookieManager` to become a `NetworkManager` data structure or to combine it into the `SiteDataManager`. I'm starting to be convinced that cookies are also site data as well. What do you think?

Would you be willing to combine the two data structures into a single `SiteDataManager`?

janvarga commented:
> Just a question: does it makes sense for the `CookieManager` to become a `NetworkManager` data structure or to combine it into the `SiteDataManager`. I'm starting to be convinced that cookies are also site data as well. What do you think?
> 
> Would you be willing to combine the two data structures into a single `SiteDataManager`?

Yeah, now that I think about it again, it probably makes more sense to have just one manager. I was likely too influenced by WPE WebKit. That said, the planned protocol handlers probably do deserve a separate `NetworkManager`.

How about a new follow-up PR where I rename `CookieManager` to `NetworkManager` and move `clear_cookies` to `SiteDataManager`?

janvarga committed:
libservo: Rename `CookieManager` to `NetworkManager` and move `clear_cookies` to `SiteDataManager`

This patch renames `CookieManager` to `NetworkManager` to better reflect its
future scope. It also moves `clear_cookies` to `SiteDataManager`, aligning
cookie clearing with other planned site data APIs.

No functional changes are intended.

Signed-off-by: Jan Varga <jvarga@igalia.com>

mrobinson reviewed (approved):
Looks good!

One nit to address before landing:

Let's name the Rust files after the data structure they expose, so `site_data.rs` would be `site_data_manager.rs` and `net.rs` would be `network_manager.rs`. This is just to be consistent with the API Rust files we already have.

janvarga committed:
libservo: Rename rust files to match the data structure they expose

Signed-off-by: Jan Varga <jvarga@igalia.com>

janvarga commented:
> Let's name the Rust files after the data structure they expose, so `site_data.rs` would be `site_data_manager.rs` and `net.rs` would be `network_manager.rs`. This is just to be consistent with the API Rust files we already have.

Done (in a new commit)
